### PR TITLE
[toast] Enable closing all toasts

### PR DESCRIPTION
**Pull request by chuganzy**
This PR adds a way to close all toasts at once.

The `close` API now accepts an optional `id`:
- `close(id)` closes a specific toast
- `close()` closes all active toasts

Alternatively, I considered introducing a separate `closeAll` API, but decided to keep the change minimal for the following reasons:

- Adding `closeAll` could require changes to `ToastManagerEvent` used by ` subscribe`. I was not sure whether it is a public API, and making changes to the `ToastManagerEvent` might be considered a breaking change (though that can be decoupled).
- Other libraries such as Sonner and Chakra expose the same behavior through a single `dismiss` API, where calling it without an argument dismisses all toasts.

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/base-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

**Comment by atomiks** [comment]
I'm okay with this change, but it will conflict a lot with https://github.com/mui/base-ui/pull/3464, not sure whether to merge it before or after @flaviendelangle

**Comment by chuganzy** [comment]
Given that this PR is significantly smaller than the other one, I think it may make sense for me to wait 😅
I can take care of resolving the conflicts on my end as needed 🙌

**Comment by flaviendelangle** [comment]
As you want @atomiks, I resolved all your feedback on #3464 :+1:

File: packages/react/src/toast/store.ts [hunk 1]
```diff
@@ -226,15 +226,22 @@ export class ToastStore extends ReactStore<State, {}, typeof selectors> {
     }
   };
 
-  closeToast = (toastId: string) => {
-    const toast = selectors.toast(this.state, toastId);
-    toast?.onClose?.();
-
+  closeToast = (toastId?: string) => {
+    const closeAll = toastId === undefined;
     const { limit, toasts } = this.state;
```
[the remark this answers is not in the document]
Comment (chuganzy) [reply packages/react/src/toast/store.ts:244]: It is true. This can also happen with single-close as well.
Since this is an existing issue, I will create a separate PR to address it.

**Comment by chuganzy** [comment]
@atomiks 

Quick ping on this PR 🙂
CI is failing, but it looks like it's during the initial repo clone rather than the tests, so it might not be related to this PR. Could you take a look when you have a chance? Thanks!

**Comment by chuganzy** [comment]
Uh-oh.. Prettier rule has changed. I will re-format and push the change soon!

**Comment by atomiks** [comment]
## Codex Review

### Overview
This patch adds support for dismissing every toast by calling `close()` without an ID, and wires that behavior through the manager, provider, store, docs, and tests. The implementation looks consistent overall, and I did not find a new correctness issue introduced by the API change.

### Findings
No blocking issues found in this patch.

#### 1. 🟡 [Non-blocking] Close-all exposes an existing duplicate `onClose` edge case

**Impact**: `onClose` is already not idempotent if the same toast is closed twice. This patch makes that pre-existing behavior easier to hit because `close()` with no ID also iterates over toasts that are already ending, so consumers may now encounter the duplicate callback through the new public API.

**Evidence**: `packages/react/src/toast/store.ts:245-248` calls `onClose` for every toast in the close-all branch before `packages/react/src/toast/store.ts:265-269` leaves already-ending toasts in place. I verified that `master` already double-calls `onClose` for `closeToast(id)` -> `closeToast(id)`, and this branch extends the same behavior to `closeToast(id)` -> `closeToast()`.

**Recommendation**: Consider handling this in a follow-up by only invoking `onClose` when a toast first transitions to `'ending'`, and add a regression test that covers both repeated single-close and single-close-then-close-all flows.

### Confidence: 4/5
Reviewed the full branch diff against current `master` and ran the focused `createToastManager`, `useToastManager`, and Chromium `toast` suites. The API, docs, and test wiring look aligned; the remaining concern is limited to the pre-existing callback edge case above.

### Notes
- Review scope covered the full branch diff against current `master`, not just the latest commit.
- The duplicate `onClose` behavior appears to predate this PR; this patch widens its reach rather than introducing it from scratch.

**Comment by chuganzy** [comment]
@atomiks As always, thanks for jumping in and taking care of this so quickly. Much appreciated!
